File: backend/src/main/java/org/artso/budget_manager/entity/Item.java
package org.artso.budget_manager.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Set;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @NotBlank
    private String name;
    private String description;
    @Min(0)
    private BigDecimal amount;
    private String category;

    @ManyToOne
    @JoinColumn(name = "author_id")
    private AppUser author;

    @ManyToMany
    @JoinTable(
            name = "item_shared_groups",
            joinColumns = @JoinColumn(name = "item_id"),
            inverseJoinColumns = @JoinColumn(name = "group_id")
    )
    private Set<UserGroup> sharedGroups;
}

File: frontend/lib/pages/items_list/dto/item.dart
class Item {
  final int id;
  final String name;
  final String description;
  final double amount;
  final String category;

  Item({
    required this.id,
    required this.name,
    required this.description,
    required this.amount,
    required this.category,
  });

  factory Item.fromMap(Map<String, dynamic> map) {
    return Item(
      id: map['id'],
      name: map['name'],
      description: map['description'],
      amount: map['amount'].toDouble(),
      category: map['category'],
    );
  }
}
